server/src/main.rs

```diff
@@ -70,88 +208,94 @@ async fn shutdown_signal() {
-            .with_graceful_shutdown(shutdown_signal())
-            .await
-            .unwrap();
-    });
+/// Tokio signal handler that will wait for a user to press CTRL+C.
+/// We use this in our hyper `Server` method `with_graceful_shutdown`.
+///
+/// # Arguments
+///
+/// # Examples
+///
+/// ```
+/// Server::bind(&"0.0.0.0:8000".parse().unwrap())
+/// .serve(app.into_make_service())
+/// .with_graceful_shutdown(shutdown_signal())
+/// .await
+/// .unwrap();
+/// ```
+pub async fn shutdown_signal() {
+    tokio::signal::ctrl_c()
+        .await
+        .expect("expect tokio signal ctrl-c");
+    println!("signal shutdown!");
 }
 
-#[tokio::main]
-async fn main() -> Result<(), Error> {
-    // REST API in background thread
-    rest_server_start();
-
-    // GRPC Server
-    let grpc_port = std::env::var("DOCKER_PORT_GRPC")
-        .unwrap_or_else(|_| "50051".to_string())
+/// Starts the REST API server for this microservice
+pub async fn rest_server(grpc_clients: GrpcClients) {
+    let rest_port = std::env::var("DOCKER_PORT_REST")
+        .unwrap_or_else(|_| "8000".to_string())
         .parse::<u16>()
-        .unwrap_or(50051);
+        .unwrap_or(8000);
+    #[derive(OpenApi)]
+    #[openapi(
+        paths(
+            query_flight,
+            query_vertiports,
+            confirm_flight,
+            cancel_flight
+        ),
+        components(
+            schemas(
+                rest_types::FlightOption,
+                rest_types::Vertiport,
+                rest_types::ConfirmError,
+                rest_types::VertiportsQuery,
+                rest_types::FlightQuery
+            )
+        ),
+        tags(
+            (name = "svc-cargo", description = "svc-cargo API")
+        )
+    )]
+    struct ApiDoc;
 
-    let addr = format!("[::1]:{grpc_port}").parse().unwrap();
-    let imp = CargoGrpcImpl::default();
-    let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
-    health_reporter
-        .set_serving::<CargoRpcServer<CargoGrpcImpl>>()
-        .await;
+    let app = Router::new()
+        .layer(Extension(grpc_clients))
+        // .merge(SwaggerUi::new("/swagger-ui/*tail").url("/api-doc/openapi.json", ApiDoc::openapi()))
+        .fallback(not_found.into_service())
+        .route(rest_types::ENDPOINT_CANCEL, routing::delete(cancel_flight))
+        .route(rest_types::ENDPOINT_QUERY, routing::get(query_flight))
+        .route(rest_types::ENDPOINT_CONFIRM, routing::put(confirm_flight))
+        .route(
+            rest_types::ENDPOINT_VERTIPORTS,
+            routing::get(query_vertiports),
+        );
 
-    println!("gRPC Server Listening at {}", addr);
-    tonic::transport::Server::builder()
-        .add_service(health_service)
-        .add_service(CargoRpcServer::new(imp))
-        .serve_with_shutdown(addr, shutdown_signal())
+    println!("REST API Hosted at 0.0.0.0:{rest_port}");
+    let address = SocketAddr::from((Ipv4Addr::UNSPECIFIED, rest_port));
```

Comment:
For the grpc server we used `let addr = format!("[::1]:{grpc_port}").parse().unwrap();` to create a SocketAddr. I think we should use the same for the rest server:
`let addr = format!("[::1]:{rest_port}").parse().unwrap();`

Reply:
Fixed